packages/contracts/contracts/core/component/DaoAuthorizableUpgradeable.sol

```diff
@@ -0,0 +1,64 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.10;
+
+import "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+import "@openzeppelin/contracts-upgradeable/utils/ContextUpgradeable.sol";
+
+import "../permission/PermissionManager.sol";
+import "./../IDAO.sol";
+
+/// @title DaoAuthorizableUpgradable
+/// @author Aragon Association - 2022
+/// @notice An abstract contract providing a meta transaction compatible modifier to authorize function calls through an associated DAO.
+/// This contract provides an `auth` modifier that can be applied to functions in inheriting contracts. The permission to call these functions is managed by the associated DAO.
+/// @dev Make sure to call `__DaoAuthorizable_init` during initialization of the inheriting contract.
+///      This contract is compatible with meta transactions through OZ's `ContextUpgradable`.
+abstract contract DaoAuthorizableUpgradeable is Initializable, ContextUpgradeable {
```

Comment:
Can we reduce code duplication with `DaoAuthorizabl.sol` by creating an abstract base class?